### DISCUSSION TITLE
Configure or remember Date Range for Meal Planner

### DISCUSSION THREAD

**Discussion by grasdk** [mealie-recipes/mealie#3022] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Whenever one enters the meal planner, the date range is reset.

### Please provide a concise description of the feature that would resolve your issue.

Implement planner settings where you can set
- default number of days to show in planner
- first day to show:
-- day of week
--- perhaps with added ability to select current week or current week +/- N, where N is a number.
-- day of month
-- today - implying the the calender shows a "sliding window" of days
--- possibly with added ability to select today +/- N, where N is a the number of days to show

### Please consider and list out some caveats or tradeoffs made in your design decision

Advanced options like +/- N weeks or days may be confusing, but with proper defaults and tool tips, it can probably be mitigated.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by grasdk** [comment]
Found a partial overlap here: https://github.com/mealie-recipes/mealie/discussions/1780

Solving this feature request would solve that as well.

**Comment by boilermaker-ben** [comment]
My wife and I would also be very interested in this as a feature, or even just a guide as to how to make a permanent change in the code (I'm using Docker). Seeing the previous few days/week would be great for making sure the meals being planned have better variety in addition to know when leftovers may no longer be good to eat.

**Comment by angelsafira** [comment]
I’m interested in this as a feature. I would like to be able to set the week, for example, from Thursday to Wednesday. Thank you very much.

**Comment by DevFreezing** [comment]
I find this feature quite appealing. It would be great if there was an option to configure the week to start and end on specific days, I would really appreciate it if this functionality could be implemented.

**Comment by michael-genson** [comment]
> default number of days to show in planner

This should be implemented already: https://github.com/mealie-recipes/mealie/pull/3650

> first day to show

This should be configurable in your group preferences (`https://your-site.com/group`)
